tests/cases/fourslash/autoImportPackageJsonImportsLength2.ts

```diff
@@ -0,0 +1,18 @@
+/// <reference path="fourslash.ts" />
+
+// @module: nodenext
+
+// @Filename: /package.json
+//// {
+////   "imports": {
+////     "#*": "./src/*.ts"
+////   }
+//// }
+
+// @Filename: /src/a/b/c/something.ts
+//// export function something(name: string): any;
+
+// @Filename: /a.ts
+//// something/**/
+
+verify.importFixModuleSpecifiers("", ["#a/b/c/something"]);
```

Comment:
Similar question here... `*.ts` can't resolve at runtime, correct? I had thought the other test above didn't offer `#...` up because it was invalid, but this one does?

Reply:
Like the other, this is suspicious with no other info given, but there’s no rule against it. That said, looking closer at the range of tests here, I feel like they skew toward unconventional usage, and I actually suspect the mapping from output files to input files that module resolution does for `imports` is probably missing in the module specifier generation implementation.

@emmatown, can you add some tests with `outDir` and `imports` pointing to those outputs? Let’s put those tests in `fourslash/server` and include a virtual tsconfig.json file, which will prevent any `.js` and `.d.ts` files in your test from being forced into being input files in the compilation.

Reply:
IMO, these become a bit more conventional with a `"types"` condition in front of them `.d.` in the extension path - we're just blind to what conditions are active when checking if a file's inclusion is valid or not (which is true to node itself - the cjs loader will happily _attempt_ to load an esm file and choke on it).

Still, in the interest of simplifying most of the test cases to test to smallest unit of functionality, I kinda understand wanting to write them like this, instead.